tests/Doctrine/Tests/DBAL/Functional/Driver/IBMDB2/DB2StatementTest.php

```diff
@@ -29,6 +31,7 @@ protected function setUp() : void
     public function testExecutionErrorsAreNotSuppressed() : void
     {
         $stmt = $this->connection->prepare('SELECT * FROM SYSIBM.SYSDUMMY1 WHERE \'foo\' = ?');
+        assert($stmt instanceof Statement);
```

Comment:
That link doesn't seem to deep link properly.

Clarify the return type in `Doctrine\DBAL\Connection::prepare()` ?

Reply:
I updated the link but it's not that important. Yes, given the number of assertions like this in the patch, we should make it clear that a DBAL Connection returns a DBAL Statement.

Reply:
This seems like it is out of the scope of this PR to make BC breaks in the core API of the DBAL.

Reply:
It doesn't have to be a breaking change. What I had in mind is adding a `@return` annotation which would hopefully override the type hint from the SA perspective but won't be enforced by the compiler.

Reply:
That still feels out of the scope of enabling phpstan in the tests folder. I don't mind making that change but I want to do it in an isolated PR without all these other changes muddying the review and testing of the change.

Reply:
I see your point. It's okay as long as it doesn't make you rework the tests which do work and cover valid cases like the one with `DateTime`. If you can fix those tests without changing their logic, then assertions like this can stay.